src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs

```diff
@@ -528,23 +529,24 @@ private BoundLambda ReallyBind(NamedTypeSymbol delegateType)
 
             if (lambdaSymbol.RefKind == CodeAnalysis.RefKind.RefReadOnly)
             {
-                Binder.Compilation.EnsureIsReadOnlyAttributeExists(diagnostics, lambdaSymbol.DiagnosticLocation, modifyCompilation: false);
+                compilation.EnsureIsReadOnlyAttributeExists(diagnostics, lambdaSymbol.DiagnosticLocation, modifyCompilation: false);
             }
 
             var lambdaParameters = lambdaSymbol.Parameters;
-            ParameterHelpers.EnsureIsReadOnlyAttributeExists(lambdaParameters, diagnostics, modifyCompilation: false);
+            ParameterHelpers.EnsureIsReadOnlyAttributeExists(compilation, lambdaParameters, diagnostics, modifyCompilation: false);
 
             if (returnType.HasType)
             {
-                if (returnType.NeedsNullableAttribute())
+                if (compilation.ShouldEmitNullableAttributes(lambdaSymbol) &&
+                    returnType.NeedsNullableAttribute())
```

Comment:
>returnType.NeedsNullableAttribute() [](start = 20, length = 35)

Should we stop emitting nullable attributes for any synthesized code not meant for direct consumption?

Reply:
Logged [https://github.com/dotnet/roslyn/issues/36466](https://github.com/dotnet/roslyn/issues/36466).

---
In reply to: [293925385](https://github.com/dotnet/roslyn/pull/36398#discussion_r293925385) [](ancestors = 293925385)